vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/style/TextStyle.java

```diff
@@ -0,0 +1,413 @@
+    @JsonIdentityReference(alwaysAsId = true)
+    public Fill getBackgroundFill() {
+        return backgroundFill;
+    }
+
+    /**
+     * Sets the fill color to use for the background of the text. The fill's
+     * color must be a valid CSS color. By default, the text does not use a
+     * background color.
+     *
+     * @param backgroundFill
+     *            the new background fill color
+     */
+    public void setBackgroundFill(Fill backgroundFill) {
+        removeChild(this.backgroundFill);
+        this.backgroundFill = backgroundFill;
+        addNullableChild(backgroundFill);
+    }
+
+    /**
+     * Sets the fill color to use for the background of the text. The fill's
+     * color must be a valid CSS color. By default, the text does not use a
+     * background color.
+     *
+     * @param fillColor
+     *            the new background fill color
+     */
+    public void setBackgroundFill(String fillColor) {
+        setBackgroundFill(new Fill(fillColor));
+    }
+
+    /**
+     * The stroke with which to render the texts background border.
+     *
+     * @return the text's background stroke
+     */
+    @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
+    @JsonIdentityReference(alwaysAsId = true)
+    public Stroke getBackgroundStroke() {
+        return backgroundStroke;
+    }
+
+    /**
+     * Sets the stroke with which to render the texts background border. The
+     * stroke's color must be a valid CSS color, and its width is specified in
+     * pixels. By default, the text does not use a background stroke.
+     *
+     * @param backgroundStroke
+     *            the new background stroke
+     */
+    public void setBackgroundStroke(Stroke backgroundStroke) {
+        removeChild(this.backgroundStroke);
+        this.backgroundStroke = backgroundStroke;
+        addNullableChild(backgroundStroke);
+    }
+
+    /**
+     * Sets the stroke with which to render the texts background border. The
+     * stroke's color must be a valid CSS color, and its width is specified in
+     * pixels. By default, the text does not use a background stroke.
+     *
+     * @param strokeColor
+     *            the new background stroke color
+     * @param strokeWidth
+     *            the new background stroke width
+     */
+    public void setBackgroundStroke(String strokeColor, int strokeWidth) {
+        setBackgroundStroke(new Stroke(strokeColor, strokeWidth));
+    }
+
+    /**
+     * The padding for the text's background.
+     *
+     * @return the background padding
+     */
+    public int getPadding() {
+        return padding;
+    }
+
+    /**
+     * Sets the padding for the text's background. The padding is applied to all
+     * sides of the background. Default value is {@code 0}.
+     *
+     * @param padding
+     *            the new padding
+     */
+    public void setPadding(int padding) {
```

Comment:
Should we support setting the padding for each side or, at least, for each pair of sides (top/bottom and left/right) from the start? My reason is that usually, one wants at least to have bigger horizontal padding than the vertical one.

Reply:
My plan was to keep this as simple as possible and wait and see if there is sufficient interest in this 🙂

Reply:
That was my first assumption, indeed. But, nevertheless, it still made me wonder whether it makes sense to add this API from the start. Anyway, we can add an enhancement ticket to address this behavior and wait for any reactions/requests.